jwst/wavecorr/tests/test_wavecorr.py

```diff
@@ -108,47 +108,63 @@ def test_skipped():
 
     # Test an error is raised if assign_wcs or extract_2d were not run.
     im.meta.exposure.type = 'NRS_FIXEDSLIT'
-    with pytest.raises(AttributeError):
+    with pytest.raises(ValueError):
         WavecorrStep.call(im)
 
     outa = AssignWcsStep.call(im)
+
+    # wcs and other metadata necessary for a good (not skipped) result 
+    dither = {'x_offset': 0.0, 'y_offset': 0.0}
+    outa.meta.dither = dither
+    outa.meta.wcsinfo.v3yangle = 138.78
+    outa.meta.wcsinfo.vparity = -1
+    outa.meta.wcsinfo.v2_ref = 321.87
+    outa.meta.wcsinfo.v3_ref = -477.94
+    outa.meta.wcsinfo.roll_ref = 15.1234
+    outa.meta.instrument.fixed_slit = "S400A1"
+
     oute = Extract2dStep.call(outa)
+
+    # ensure the source position is set as expected by extract2d
+    ind = np.nonzero([s.name == 'S400A1' for s in oute.slits])[0].item()
+    source_pos = (0.004938526981283373, -0.02795306204991911)
+    assert_allclose((oute.slits[ind].source_xpos, oute.slits[ind].source_ypos),source_pos)
+
     outs = SourceTypeStep.call(oute)
 
+    # primary slit is found by source type step to be EXTENDED
+    # which means its source position is reset to (0, 0)
+    # hack it here to be a point source with the same offset as before here
+    outs.slits[ind].source_type = 'POINT'
+    outs.slits[ind].source_xpos = source_pos[0]
+    outs.slits[ind].source_ypos = source_pos[1]
+
     # Test step is skipped if no coverage in CRDS
-    outs.slits[0].meta.observation.date = '2001-08-03'
+    outs.slits[ind].meta.observation.date = '2001-08-03'
     outw = WavecorrStep.call(outs)
     assert out.meta.cal_step.wavecorr == "SKIPPED"
```

Comment:
It looks like this test never did what the author thought it did, since it's putting the result of `wavecorr` into `outw` but querying `out` in the assert statement.  After changing this, the check fails; `get_reference_file` seems to have no problem finding reference files, and the step completes instead of hitting the `if` statement that checks if the reference file is N/A.  While this isn't really directly related to this PR, it definitely looks like it should be fixed, and IMO it's not worth making a new ticket just for this.

One guess as to why this is happening is that setting the slit's observation date metadata attribute is not the correct date to change in the test.
 @hbushouse or @melanieclarke do you know what date/time information is requested when `get_reference_file` is called?

Reply:
I think it just needs to set that date in the top-level meta.observation.date:
`outs.meta.observation.date = '2001-08-03'`